# Ensure major(), minor() and makedev() are defined
# Fix for DSM 7.2 having glibc 2.36 and SRM 1.3 having glibc 2.32
# 
# New versions of glibc (since 2.28) and musl (since 1.1.23) actually no longer provide a definition 
# of major by including sys/types.h.
# It's a bit unfortunate since on Linux, we now need to include sys/sysmacros.h
# 
# Solution based on https://github.com/kristapsdz/openrsync/issues/13#issuecomment-559931285
# 
--- src/shared/macro.h.orig	2015-02-16 20:55:07.000000000 +0000
+++ src/shared/macro.h	2025-09-21 20:19:43.838878204 +0000
@@ -24,6 +24,10 @@
 #include <assert.h>
 #include <sys/param.h>
 #include <sys/types.h>
+#ifndef major
+// for the case that sys/sysmacros.h is not included within sys/types.h
+#include <sys/sysmacros.h>
+#endif
 #include <sys/uio.h>
 #include <inttypes.h>
 
